main.c

```diff
@@ -1057,14 +1061,27 @@ int main(int argc, char **argv)
 	if (tcmu_setup_log())
 		goto destroy_config;
 
-	tcmu_dbg("handler path: %s\n", handler_path);
+	fd = creat(TCMU_LOCK_FILE, S_IRUSR | S_IWUSR);
+	if (fd == -1) {
+		tcmu_err("creat(%s) failed: [%m]\n", TCMU_LOCK_FILE);
+		goto destroy_log;
+	}
+
+	lock.l_type = F_WRLCK;
+	if (fcntl(fd, F_SETLK, &lock) == -1) {
+		tcmu_err("tcmu-runner is already running...\n");
```

Comment:
Should we check that only when errno == EACCES or EAGAIN we will be sure that the tcmu-runner main process is already runing ? Or it should be other op errors.

Reply:
Sure, I will defend on EAGAIN